react-components/src/components/RuleBasedOutputs/RuleBasedOutputsSelector.tsx

```diff
@@ -31,6 +37,23 @@ export function RuleBasedOutputsSelector({
     fetchNextPage
   } = useAllMappedEquipmentAssetMappings(models);
 
+  const contextualizedAssetNodes =
+    assetMappings?.pages
+      .flat()
+      .flatMap((item) => item.assets)
+      .map(convertAssetMetadataKeysToLowerCase) ?? [];
+
+  const expressions = ruleSet?.rulesWithOutputs
+    .map((ruleSet) => ruleSet.rule.expression)
+    .filter(isDefined);
+  const timeseriesExternalIdsFromRule = traverseExpressionToGetTimeseries(expressions) ?? [];
+
+  const { isLoading: isLoadingAssetIdsAndTimeseriesData, data: assetIdsWithTimeseriesData } =
+    useAssetsAndTimeseriesLinkageDataQuery({
+      timeseriesExternalIdsFromRule,
+      contextualizedAssetNodes
+    });
+
```

Comment:
There is a `useEffect` below that should be moved up since it's associated with `useAllMappedEquipmentAssetMappings()`

Reply:
You mean closer to the useAllMappedEquipmentAssetMappings ?
For design patterns/convention, Is it not better to keep all useEffects at the end of the component and right before the jsx part?
But I changed here:
https://github.com/cognitedata/reveal/pull/4414/commits/82528c8036f90d3a908a0dc63f1693254f5455bd

Reply:
Aah, I see now that that's the convention you're using. I think I'd still prefer it the new way, especially since it's not clear from the names used inside the `useEffect` what is being fetched

Reply:
So you prefer to move closer to the associated place? If so, then It is done already in the commit above.